File: app/src/test/java/com/nytimes/android/external/playbillingtester/TestSchedulerProvider.java
package com.nytimes.android.external.playbillingtester;


import com.nytimes.android.external.playbillingtester.di.SchedulerProvider;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.internal.schedulers.ExecutorScheduler;


public class TestSchedulerProvider implements SchedulerProvider {

    private final Scheduler immediate = new Scheduler() {
        @Override
        public Worker createWorker() {
            return new ExecutorScheduler.ExecutorWorker(Runnable::run);
        }
    };

    @Override
    public Scheduler getMainThread() {
        return AndroidSchedulers.mainThread();
    }

    @Override
    public Scheduler getComputationThread() {
        return immediate;
    }

    @Override 
    public Scheduler getIoThread() {
        return immediate;
    }
}

File: app/src/test/java/com/nytimes/android/external/register/TestSchedulerProvider.kt
package com.nytimes.android.external.register

import com.nytimes.android.external.register.di.SchedulerProvider

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.internal.schedulers.ExecutorScheduler

class TestSchedulerProvider : SchedulerProvider {

    private val immediate = object : Scheduler() {
        override fun createWorker(): Scheduler.Worker {
            return ExecutorScheduler.ExecutorWorker { it.run() }
        }
    }

    override fun getMainThread(): Scheduler {
        return AndroidSchedulers.mainThread()
    }

    override fun getComputationThread(): Scheduler {
        return immediate
    }

    override fun getIoThread(): Scheduler {
        return immediate
    }
}
